subsys/bluetooth/host/adv.c

```diff
@@ -1360,18 +1373,22 @@ int bt_le_ext_adv_delete(struct bt_le_ext_adv *adv)
 static void adv_timeout(struct k_work *work)
 {
 	int err = 0;
-#if defined(CONFIG_BT_EXT_ADV)
+
 	struct k_work_delayable *dwork;
 	struct bt_le_ext_adv *adv;
 
 	dwork = k_work_delayable_from_work(work);
 	adv = CONTAINER_OF(dwork, struct bt_le_ext_adv, timeout_work);
 
-	if (atomic_test_bit(adv->flags, BT_ADV_EXT_ADV)) {
+#if defined(CONFIG_BT_EXT_ADV)
+	if (atomic_test_bit(adv->flags, BT_ADV_EXT_ADV) &&
```

Comment:
This check is incorrect. You should call bt_le_adv_stop() if `adv == bt_dev.adv` (Meaning it is the legacy advertiser and was started with bt_le_adv_start()), otherwise use bt_le_ext_adv_stop() (because it was started with bt_le_ext_adv_start()).

Please make this separate commit as it is a different problem than the timeout fix.